src/main.rs

```diff
@@ -13,13 +13,40 @@ use crate::inputs::{request_inputs, validate_inputs};
 use crate::trusted_dealer_keygen::trusted_dealer_keygen;
 
 fn main() -> io::Result<()> {
-    let config = request_inputs();
+    let mut reader = Box::new(io::stdin().lock());
+    let config = request_inputs(&mut reader);
+    match config {
+        Ok(_) => (),
+        Err(e) => {
+            eprintln!("Error: {}", e);
+            std::process::exit(exitcode::DATAERR)
+        }
+    }
```

Comment:
Slightier less verbose to do something like

```suggestion
    if let Err(e) = config {
        eprintln!("Error: {}", e);
        std::process::exit(exitcode::DATAERR)
    }
```

(this one I did test :laughing: )

It's really usually best to use `match` in enums to make sure that if new entries are added, then it will fail to build until they're handled. But `Result` won't likely change so it's fine to use this instead.

Reply:
Added in [ad3c315](https://github.com/ZcashFoundation/frost-zcash-demo/pull/4/commits/ad3c31588ba00a06aae1129cad967403e04c5fe7)